stdlib/database/sql/src/main/java/org/ballerinalang/database/sql/statement/AbstractSQLStatement.java

```diff
@@ -404,28 +407,30 @@ protected void cleanupResources(Statement stmt, Connection conn, boolean connect
                 conn.close();
             }
         } catch (SQLException e) {
-            throw new BallerinaException("error in cleaning sql resources: " + e.getMessage(), e);
+            throw new Error(errorMessagePrefix + "error in cleaning sql resources: " + e.getMessage(), e);
         }
     }
 
     /**
      * This will close database connection, statement and the resultset.
      *
+     * @param errorMessagePrefix Error message prefix
      * @param rs   SQL resultset
      * @param stmt SQL statement
      * @param conn SQL connection
      * @param connectionClosable Whether the connection is closable or not. If the connection is not closable this
      * method will not release the connection. Therefore to avoid connection leaks it should have been taken care
      * of externally.
      */
-    protected void cleanupResources(ResultSet rs, Statement stmt, Connection conn, boolean connectionClosable) {
+    protected void cleanupResources(String errorMessagePrefix, ResultSet rs, Statement stmt, Connection conn,
+                                    boolean connectionClosable) {
         try {
             if (rs != null && !rs.isClosed()) {
                 rs.close();
             }
-            cleanupResources(stmt, conn, connectionClosable);
+            cleanupResources(errorMessagePrefix, stmt, conn, connectionClosable);
         } catch (SQLException e) {
-            throw new BallerinaException("error in cleaning sql resources: " + e.getMessage(), e);
+            throw new Error(errorMessagePrefix + "error in cleaning sql resources: " + e.getMessage(), e);
```

Comment:
Why we need to throw an `Error` here? Since this `cleanupResources` method is getting called within catch blocks and finally blocks we need to throw `BallerinaException` here to make it a panic scenario.